olive/passes/olive_pass.py

```diff
@@ -382,27 +382,20 @@ def run(
                 rank_output_path = Path(output_model_path).with_suffix("") / str(rank)
                 output_rank_model = self._run_for_config(input_rank_model, data_root, config, rank_output_path)
                 output_filepaths.append(output_rank_model.model_path)
-            return DistributedOnnxModel(output_filepaths, inference_settings=model.inference_settings)
+            output_model = DistributedOnnxModel(output_filepaths, inference_settings=model.inference_settings)
         elif isinstance(model, CompositeOnnxModel) and not self._accepts_composite_model:
             components = []
             component_names = []
             for cidx, child in enumerate(model.get_model_components()):
                 component_output_path = Path(output_model_path).with_suffix("") / str(cidx)
                 output_model_components = self._run_for_config(child, data_root, config, str(component_output_path))
-                components.append(self.inherit_hf_config_from_input_model(model, output_model_components))
+                output_model_components.extend_model_attributes(child.model_attributes)
+                components.append(output_model_components)
                 component_names.append(model.get_model_component_name(cidx))
-            return CompositeOnnxModel(components, component_names, hf_config=model.hf_config)
+            output_model = CompositeOnnxModel(components, component_names)
         else:
             output_model = self._run_for_config(model, data_root, config, output_model_path)
-            return self.inherit_hf_config_from_input_model(model, output_model)
-
-    def inherit_hf_config_from_input_model(self, input_model: OliveModel, output_model: OliveModel) -> OliveModel:
-        # TODO: handle the case with local model path and huggingface model config for torch model
-        if hasattr(output_model, "hf_config") and not isinstance(output_model, PyTorchModel):
-            # not all models have hf_config
-            # Do not inherit hf_config from input model if output_model is PyTorchModel for time being.
-            if not output_model.hf_config and getattr(input_model, "hf_config", None):
-                output_model.hf_config = input_model.hf_config
+        output_model.extend_model_attributes(model.model_attributes)
         return output_model
```

Comment:
Here, I am not sure how a pass is able to update the model_attributes.
I think, this should be pushed inside __run_for_confg()

Reply:
`extend_model_attributes` will try merge output model's attributes and input model's attributes. When there is conflict, output model's attribute is always the first choice as pass already modified it.

Reply:
Updated with following logics:
1. we assume that the pass should return complete config even they modified it. That says the update logic is pushed into _run_for_config.
2. if output model's attributes is empty, we use the one in input model.